### ISSUE TITLE
OC - Error Message Localization

### ISSUE THREAD

**Issue by b-k-white** [tsgrp/HPI#1060]
Currently we have the ability to localize messages and labels in HPI so that they can be translated into multiple languages or client preferences. However, messages that bubble up from OC do not currently adhere to this standard. In order for OC error messages to bubble up and become localized, we could send a message from OC that keys to a localized messaged, but this would make OC error messages difficult to understand when reading or fixing code. I'm wondering if there is a better alternative to the above, or if we should just forgo having OC messages from being localized.

Image explanation:

![image](https://cloud.githubusercontent.com/assets/8796489/14359065/6152243e-fcb5-11e5-93fe-cfb8b3c4587c.png)

**Comment by gsteimer** [comment]
The way to do this would be to integrate Java I18N into OC.  Resource bundles can be internationalized in Java without issue. See: https://docs.oracle.com/javase/tutorial/i18n/intro/quick.html

I believe Spring also has some I18N capabilities, but it may be more tied into Spring MVC, not sure.

The trick is going to be knowing which locale is selected on the HPI side...

**Comment by dgrumieaux** [comment]
Use request headers? I think those exist most of the time..

**Comment by gsteimer** [comment]
@dgrumieaux - yeah, I was thinking that or a cookie would work.  But there's also the question of how we get that value from the REST endpoint to the actual `*Impl` code that throws the error.  I'm thinking we probably don't want to add a `String locale` param to every OC method...

**Comment by dgrumieaux** [comment]
Yeah that sounds terrible to refactor, plus wordy. Should probably research
what other java codebases do for this
